bindings/java/src/org/sleuthkit/datamodel/OsAccountRealmManager.java

```
@@ -0,0 +1,435 @@
+	 * @param nameType Name type.
+	 * 
+	 * @return OsAccountRealm
+	 * @throws TskCoreException 
+	 */
+	OsAccountRealm updateRealmName(long realmId, String realmName, OsAccountRealm.RealmNameType nameType) throws TskCoreException {
+		
+		db.acquireSingleUserCaseWriteLock();
+		try (CaseDbConnection connection = db.getConnection())  {
+			String updateSQL = "UPDATE tsk_os_account_realms SET name = ?, name_type = ? WHERE id = ?";
+			PreparedStatement preparedStatement = connection.getPreparedStatement(updateSQL, Statement.NO_GENERATED_KEYS);
+			preparedStatement.clearParameters();
+
+			preparedStatement.setString(1, realmName);
+			preparedStatement.setInt(2, nameType.getId());
+			preparedStatement.setLong(3, realmId);
+			
+			connection.executeUpdate(preparedStatement);
+
+			return getRealm(realmId, connection );
+		} catch (SQLException ex) {
+			LOGGER.log(Level.SEVERE, null, ex);
+			throw new TskCoreException(String.format("Error updating realm with name = %s, id = %d", realmName, realmId), ex);
+		} finally {
+			db.releaseSingleUserCaseWriteLock();
+		}
+	}
+	
+	private final String REALM_QUERY_STRING = "SELECT realms.id as realm_id, realms.name as realm_name,"
+			+ " realms.realm_addr as realm_addr, realms.host_id, realms.name_type, "
+			+ " hosts.id, hosts.name as host_name "
+			+ " FROM tsk_os_account_realms as realms"
+			+ "		LEFT JOIN tsk_hosts as hosts"
+			+ " ON realms.host_id = hosts.id";
+	
+	/**
+	 * Get the realm from the given row id. 
+	 * 
+	 * @param id Realm row id.
+	 * @param connection Database connection to use.
+	 * 
+	 * @return Realm. 
+	 * @throws TskCoreException 
+	 */
+	OsAccountRealm getRealm(long id, CaseDbConnection connection) throws TskCoreException {
+		
+		String queryString = REALM_QUERY_STRING
+					+ " WHERE realms.id = " + id;
+		
+		try (	Statement s = connection.createStatement();
+				ResultSet rs = connection.executeQuery(s, queryString)) {
+			OsAccountRealm accountRealm = null;
```

Throw an exception instead of return null? Make this an Optional instead of return null?

---

Throw an exception here.